packages/react/src/model-context/tool.ts

```
@@ -1,26 +1,17 @@
-export type inferParameters<PARAMETERS extends Tool<any, any>["parameters"]> =
-  PARAMETERS extends Schema<any>
-    ? PARAMETERS["_type"]
-    : PARAMETERS extends z.ZodTypeAny
-      ? z.infer<PARAMETERS>
-      : never;
-
-export function tool<
-  TArgs extends Tool<any, any>["parameters"],
-  TResult = any,
->(tool: {
+export function tool<TArgs extends unknown, TResult = any>(tool: {
```

Biome (1.9.4)

[error] Constraining a type parameter to any or unknown is useless.

All types are subtypes of any and unknown.
Safe fix: Remove the constraint.


(lint/complexity/noUselessTypeConstraint)